packages/resolve/src/request-resolver.ts

```diff
@@ -43,23 +75,20 @@ export function createRequestResolver(options: IRequestResolverOptions): Request
     }
   }
 
+  function* fileOrDirIndexRequestPaths(targetPath: string) {
+    yield* fileRequestPaths(targetPath);
+    yield* fileRequestPaths(join(targetPath, 'index'));
+  }
+
   function* directoryRequestPaths(directoryPath: string) {
     if (!directoryExistsSync(directoryPath)) {
       return;
     }
-    const packageJsonPath = join(directoryPath, 'package.json');
-    const packageJson = safeReadJsonFileSync(packageJsonPath) as PackageJson;
-    const mainField = packageJson?.main;
-    const browserField = packageJson?.browser;
-
-    if (target === 'browser' && typeof browserField === 'string') {
-      const targetPath = join(directoryPath, browserField);
-      yield* fileRequestPaths(targetPath);
-      yield* fileRequestPaths(join(targetPath, 'index'));
-    } else if (typeof mainField === 'string') {
-      const targetPath = join(directoryPath, mainField);
-      yield* fileRequestPaths(targetPath);
-      yield* fileRequestPaths(join(targetPath, 'index'));
+    const resolvedPackageJson = loadPackageJsonFrom(directoryPath);
+    const mainPath = resolvedPackageJson?.mainPath;
+
+    if (mainPath !== undefined) {
```

Comment:
why `mainPath !== undefined` and not just `if(mainPath)`

Reply:
empty string is valid